domain/model/service/modelservice_test.go

```diff
@@ -10,28 +10,129 @@ import (
 	"github.com/juju/clock/testclock"
 	"github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
+	"go.uber.org/mock/gomock"
 	gc "gopkg.in/check.v1"
 
+	"github.com/juju/juju/core/constraints"
 	corecredential "github.com/juju/juju/core/credential"
 	coremodel "github.com/juju/juju/core/model"
 	modeltesting "github.com/juju/juju/core/model/testing"
 	corestatus "github.com/juju/juju/core/status"
 	"github.com/juju/juju/domain/model"
 	modelerrors "github.com/juju/juju/domain/model/errors"
+	networkerrors "github.com/juju/juju/domain/network/errors"
 	"github.com/juju/juju/internal/uuid"
 )
 
 type modelServiceSuite struct {
 	testing.IsolationSuite
 
+	mockModelState *MockModelState
+}
+
```

Comment:
It would be nice to have comments above some of the tests explaining the purpose of the test and not relying on the name to make sense to a Juju developer in 10 years ago. i.e we are testing the following scenario to ensure this part of the interface is maintained and working.

Reply:
Done some of these but don't want to take the spirit of this work away from the Author.